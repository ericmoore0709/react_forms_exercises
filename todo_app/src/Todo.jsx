import PropTypes from 'prop-types';

const Todo = ({ id, task, removeTodo }) => {

    const handleClick = (e) => {
        const todo = e.target.parentNode;
        removeTodo(todo.id);
    }

    return (
        <div id={id}>
            {task}
            <button onClick={handleClick}>X</button>
        </div>
    )
}

Todo.propTypes = {
    id: PropTypes.string.isRequired,
    task: PropTypes.string.isRequired,
    removeTodo: PropTypes.func.isRequired
}

export default Todo;