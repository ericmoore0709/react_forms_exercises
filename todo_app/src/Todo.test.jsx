import { expect, it, vi } from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

it('should not smoke', () => {
    render(<Todo id="1" task="Test task" removeTodo={vi.fn()} />);
});

it('should match snapshot', () => {
    const { asFragment } = render(<Todo id="1" task="Test task" removeTodo={vi.fn()} />);
    expect(asFragment()).toMatchSnapshot();
});

it('should fire removeTodo when X is clicked', () => {
    const removeMock = vi.fn();
    const { container } = render(<Todo id="1" task="Test task" removeTodo={removeMock} />);
    const button = container.querySelector('button');
    fireEvent.click(button);
    expect(removeMock).toBeCalledWith('1');
});
