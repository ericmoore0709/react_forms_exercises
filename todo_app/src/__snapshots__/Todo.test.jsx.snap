// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    id="1"
  >
    Test task
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
